### PR TITLE
Fix: #9 -  Fix/xhr+x local

### PR DESCRIPTION
**Pull request by m-abs**
Solves a number of problems with the webview, including #9 

* `x-local` + `XHR` stopped working on `iOS`
* `fetch` wasn't supported in all `WebView` and broken for `x-local` on Android.
* Rejection and errors in`WebViewExt.executePromise(...)` wasn't handled correctly. 
* Enables `AllowUniversalAccessFromFileURLs` on all platforms.
* iOS: Podfile ddn't specify iOS 9 as minimum requirement.
* iOS: Native-layer Podfile required unused `Alamofire` and specified wrong ios version.
* Cleaner separation between `UIWebView` and `WkWebView` implementation.

### CODE REVIEW

File: src/webview-ext.wkwebview.ts [hunk 1]
```diff
@@ -144,3 +146,356 @@ export class WKScriptMessageHandlerImpl extends NSObject implements WKScriptMess
...
+
+        const configuration = WKWebViewConfiguration.new();
+        this.wkWebViewConfiguration = configuration;
+
+        const messageHandler = WKScriptMessageHandlerImpl.initWithOwner(this.owner);
+        const wkUController = (this.wkUserContentController = WKUserContentController.new());
+        wkUController.addScriptMessageHandlerName(messageHandler, "nsBridge");
+        configuration.userContentController = wkUController;
+        configuration.preferences.setValueForKey(true, "allowFileAccessFromFileURLs");
+        configuration.setValueForKey(true, "allowUniversalAccessFromFileURLs");
```
Comment (ddfreiling) [inline src/webview-ext.wkwebview.ts]: We're setting 2 non-public variables here. Should probably wrap in a try-catch to not outright crash if this gets changed down the road.

File: src/webview-ext.wkwebview.ts [hunk 2]
```diff
@@ -144,3 +146,356 @@ export class WKScriptMessageHandlerImpl extends NSObject implements WKScriptMess
...
+    public loadUrl(src: string) {
+        const owner = this.owner.get();
+        if (!owner) {
+            return;
+        }
+
+        const ios = this.ios;
+        if (!ios) {
+            return;
+        }
```
Comment (ddfreiling) [inline src/webview-ext.wkwebview.ts]: A little verbose, perhaps just check `(!owner || !ios)` (repeated a couple of other places).

File: README.md [hunk 3]
```diff
@@ -90,18 +96,14 @@ Note: The bridge will only be available `DOMContentLoaded` or `onload` inside th
 
 ## Possible features to come:
 
-* Cookie helpers
-* Setting view-port metadata
+* Cookie helpers?
+* Setting view-port metadata?
```
Comment (ddfreiling) [inline README.md]: This has been implemented.

File: demo/app/app.ts [hunk 4]
```diff
@@ -1,7 +1,10 @@
-﻿import "nativescript-tslib";
+﻿/// <reference path="../../src/platforms/ios/GCDWebServer.d.ts" />
```
Comment (ddfreiling) [inline demo/app/app.ts]: Unused reference.

File: src/webview-ext.wkwebview.ts (hunk 1 above)
Comment (m-abs) [reply src/webview-ext.wkwebview.ts]: I can add it, but try/catch didn't catch the error from the native-layer when I tried using a non-existing variable.

**Pull request merged**